pkg/network/ovn_kubernetes.go

```diff
@@ -550,6 +585,38 @@ func shouldUpdateOVNKonIPFamilyChange(existingNode, existingMaster *appsv1.Daemo
 	return true, true
 }
 
+// shouldUpdateOVNKonPrepull implements a simple pre-pulling daemonset. It ensures the ovn-k
+// container image is (probably) already pulled by every node.
+// If the existing node daemonset has a different version then what we would like to apply, we first
+// roll out a no-op daemonset. Then, when that has rolled out to 100% of the cluster or has stopped
+// progressing, proceed with the node upgrade.
+func shouldUpdateOVNKonPrepull(existingNode, prePuller *appsv1.DaemonSet, releaseVersion string) (updateNode, renderPrepull bool) {
+	// Fresh cluster - full steam ahead! No need to wait for pre-puller.
+	if existingNode == nil {
+		return true, false
+	}
+
+	// if node is already upgraded, then no need to pre-pull
+	existingVersion := existingNode.GetAnnotations()["release.openshift.io/version"]
+	if existingVersion == releaseVersion {
+		return true, false
+	}
+
+	// at this point, we've determined we need an upgrade
+	if prePuller == nil {
+		klog.Infof("Rolling out the no-op prepuller daemonset...")
+		return false, true
+	}
+
```

Comment:
We should also check to see that the version of the pre-puller matches the expectedVersion. Unlikely, but it could happen if we upgrade and downgrade.

Reply:
Yea I have added this in the new PR. PTAL!